pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/CommentDefaultAccessModifierRule.java

```diff
@@ -158,16 +147,22 @@ private boolean isMissingComment(AccessNode decl) {
         return decl.getVisibility() == Visibility.V_PACKAGE
             // if is a default access modifier check if there is a comment
             // in this line
-            && !interestingLineNumberComments.contains(decl.getBeginLine());
+            && !hasOkComment(decl);
     }
 
     private boolean isNotIgnored(AccessNode decl) {
         return getProperty(IGNORED_ANNOTS).stream().noneMatch(decl::isAnnotationPresent);
     }
 
+    private boolean hasOkComment(AccessNode node) {
+        Pattern regex = getProperty(REGEX_DESCRIPTOR);
+        return JavaComment.getLeadingComments(node)
+                          .anyMatch(it -> regex.matcher(it.getText()).matches());
+    }
+
     private boolean shouldReportTypeDeclaration(ASTAnyTypeDeclaration decl) {
         // don't report on interfaces
-        return !decl.isRegularInterface()
+        return !(decl.isRegularInterface() && !decl.isAnnotation())
```

Comment:
this could be replaced with `!decl.isInterface()`. I don't think it makes sense to whitelist interfaces though. If they have package visibility then they should be reported. Maybe this was originally an attempt to whitelist interface members, because they're are all public anyway...